src/wp-includes/media.php

```diff
@@ -1958,6 +1971,104 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds optimization attributes to an `img` HTML tag.
+ *
+ * @since 6.3.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ *
+ * @return string Converted `img` tag with optimization attributes added.
+ */
+function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	$width              = preg_match( '/ width="([0-9]+)"/', $image, $match_width ) ? (int) $match_width[1] : '';
+	$height             = preg_match( '/ height="([0-9]+)"/', $image, $match_height ) ? (int) $match_height[1] : '';
+	$optimization_attrs = wp_get_loading_optimization_attributes(
+		'img',
+		array(
+			'width'  => $width,
+			'height' => $height,
+		),
+		$context
+	);
+
+	/**
+	 * Filters the loading optimization attributes to be added to an image.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string|bool $optimization_attrs   Attribute and it's values obtained from `wp_get_loading_optimization_attributes` function.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$optimization_attrs = apply_filters( 'wp_img_tag_add_loading_optimization_attrs', $optimization_attrs, $image, $context );
+
+	/**
+	 * Filters the `loading` attribute value to add to an image. Default `lazy`.
+	 * This filter is added in for backward compatibility.
+	 *
+	 * Returning `false` or an empty string will not add the attribute.
+	 * Returning `true` will add the default value.
+	 * `true` and `false` usage supported for backward compatibility.
+	 *
+	 * @param string|bool $optimization_attrs['loading'] Current value for `loading` attribute for the image.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$optimization_attrs['loading'] = apply_filters(
```

Comment:
I have kept `wp_img_tag_add_loading_attr` as some of the test cases were dependent on it. Though not having it in 6.3.0 won't cause any visible issue to a site, but any functionality built over it will become void. 
So to support such plugins or themes I have introduced back the filter here.

cc: @felixarntz , @joemcgill